source/material_model/interface.cc

```diff
@@ -407,6 +411,25 @@ namespace aspect
 
 
 
+    template <int dim>
+    bool
+    MaterialModelInputs<dim>::requests_property(const MaterialProperties::Property &property) const
+    {
+      //TODO: Remove this once all callers set requested_properties correctly
+      if ((property & MaterialProperties::Property::viscosity) != 0)
+        return (strain_rate.size() != 0);
+
+      //TODO: Remove this once all callers set requested_properties correctly
+      if ((property & MaterialProperties::Property::reaction_terms) != 0)
+        return (strain_rate.size() != 0);
+
+      // Note that this means 'requested_properties' can include other properties than
+      // just 'property', but in any case it at least requests 'property'.
+      return (requested_properties & property) == property;
```

Comment:
This is a bit limiting because you can not use this to see if a or b is needed: ``in.requests_property(a|b)``. If we make the member public, we won't even need this function...

Reply:
True I had not thought of this case. Still I would like to keep the function, because using `requested_properties` directly requires you to understand bitwise operators, which are a pain. Also I can nicely handle the backwards compatibility issues inside the function (because the implementation is in one place instead of 30 material models). It is not important to me to keep it inside the struct though. If we make the variable public it might as well be a MaterialModel::Utilities function.

Reply:
>  Still I would like to keep the function

true, I agree. Keeping it in the class here makes sense as well.

Reply:
what about making the last line ``!=0`` instead of ``== property``? Then one can pass ``a|b`` as well.